Mixpanel/SelectorEvaluator.m

```diff
@@ -0,0 +1,761 @@
+//
+//  SelectorEvaluator.m
+//  Mixpanel
+//
+//  Created by Madhu Palani on 3/15/19.
+//  Copyright © 2019 Mixpanel. All rights reserved.
+//
+
+#import "SelectorEvaluator.h"
+
+NSString* const errDomain = @"SelectorEvaluatorError";
+
+NSString* const OPERATOR_KEY = @"operator";
+NSString* const CHILDREN_KEY = @"children";
+NSString* const PROPERTY_KEY = @"property";
+NSString* const VALUE_KEY = @"value";
+NSString* const EVENT_KEY = @"event";
+NSString* const LITERAL_KEY = @"literal";
+NSString* const WINDOW_KEY = @"window";
+NSString* const UNIT_KEY = @"unit";
+NSString* const HOUR_KEY = @"hour";
+NSString* const DAY_KEY = @"day";
+NSString* const WEEK_KEY = @"week";
+NSString* const MONTH_KEY = @"month";
+// Typecast operators
+NSString* const BOOLEAN_OPERATOR = @"boolean";
+NSString* const DATETIME_OPERATOR = @"datetime";
+NSString* const LIST_OPERATOR = @"list";
+NSString* const NUMBER_OPERATOR = @"number";
+NSString* const STRING_OPERATOR = @"string";
+// Binary operators
+NSString* const AND_OPERATOR = @"and";
+NSString* const OR_OPERATOR = @"or";
+NSString* const IN_OPERATOR = @"in";
+NSString* const NOT_IN_OPERATOR = @"not in";
+NSString* const PLUS_OPERATOR = @"+";
+NSString* const MINUS_OPERATOR = @"-";
+NSString* const MUL_OPERATOR = @"*";
+NSString* const DIV_OPERATOR = @"/";
+NSString* const MOD_OPERATOR = @"%";
+NSString* const EQUALS_OPERATOR = @"==";
+NSString* const NOT_EQUALS_OPERATOR = @"!=";
+NSString* const GREATER_THAN_OPERATOR = @">";
+NSString* const GREATER_THAN_EQUAL_OPERATOR = @">=";
+NSString* const LESS_THAN_OPERATOR = @"<";
+NSString* const LESS_THAN_EQUAL_OPERATOR = @"<=";
+// Unary operators
+NSString* const NOT_OPERATOR = @"not";
+NSString* const DEFINED_OPERATOR = @"defined";
+NSString* const NOT_DEFINED_OPERATOR = @"not defined";
+// Special words
+NSString* const NOW_LITERAL = @"now";
+
+int const LEFT = 0;
```

Comment:
in Objective C, we usually want to use `NSInteger` rather than `int`, more detail: https://stackoverflow.com/questions/4445173/when-to-use-nsinteger-vs-int